autorest/azure/auth/authfile.go

```diff
@@ -0,0 +1,124 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"unicode/utf16"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/dimchansky/utfbom"
```

Comment:
Do we feel that this is a good package to take a dependency on?

Reply:
There was already a dependency on this package somewhere else on this repo

Reply:
For what it's worth, I added this as a registered open source dependency a couple of months ago when Tom added it.